From 1c31892c2f0efb7952aa3b8360398b2d562854d0 Mon Sep 17 00:00:00 2001
From: Marian-Cristian Rotariu <marian.rotariu@freescale.com>
Date: Mon, 9 Feb 2015 16:51:23 +0200
Subject: [PATCH 152/451] dpaa_eth: memset the local declared array

For some reason, in the simulator, the newly declared local array does not have
the value set on 0. This patch will memset the array to 0. These values are
sent to the BMan driver.

Signed-off-by: Marian-Cristian Rotariu <marian.rotariu@freescale.com>
Change-Id: I1a8c9c48748885e021d7d45ce3286e23a4e3c39d
Reviewed-on: http://git.am.freescale.net:8181/30393
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Madalin-Cristian Bucur <madalin.bucur@freescale.com>
Reviewed-by: Yang Li <LeoLi@freescale.com>
[Xulin: Original patch taken from
Linux-LS1043A-SDK-V0.4-SOURCE-20150826-yocto.iso]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/net/ethernet/freescale/dpa/dpaa_eth_sg.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth_sg.c b/drivers/net/ethernet/freescale/dpa/dpaa_eth_sg.c
index 1c56fb6..3e47bf7 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth_sg.c
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth_sg.c
@@ -102,6 +102,8 @@ static int _dpa_bp_add_8_bufs(const struct dpa_bp *dpa_bp)
 	struct device *dev = dpa_bp->dev;
 	struct sk_buff *skb, **skbh;
 
+	memset(bmb, 0, sizeof(struct bm_buffer) * 8);
+
 	for (i = 0; i < 8; i++) {
 		/* We'll prepend the skb back-pointer; can't use the DPA
 		 * priv space, because FMan will overwrite it (from offset 0)
-- 
1.7.5.4

